python/cuml/cuml/manifold/umap.pyx

```diff
@@ -845,6 +849,35 @@ class UMAP(UniversalBase,
         del X_m
         return embedding
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+
+        if "handle" in state:
+            del state["handle"]
+
+        state['_n_neighbors'] = self.n_neighbors
+        state['_a'] = self.a
+        state['_b'] = self.b
+        state['_initial_alpha'] = self.learning_rate
+        state['_disconnection_distance'] = DISCONNECTION_DISTANCES.get(self.metric, np.inf)
+
+        if hasattr(self, 'knn_dists') and hasattr(self, 'knn_indices'):
+            state['_knn_dists'] = self.knn_dists.to_output('numpy')
+            state['_knn_indices'] = self.knn_indices.to_output('numpy')
+            state['_knn_search_index'] = None
+        elif hasattr(self, '_raw_data'):
+            host_raw_data = self._raw_data.to_output('numpy')
+            state['_knn_dists'], state['_knn_indices'], state['_knn_search_index'] = \
+                nearest_neighbors(host_raw_data, self.n_neighbors, self.metric,
+                                  self.metric_kwds, False, self.random_state)
```

Comment:
I think it should be good, but if there's a possibility to avoid running the KNN, that would be ideal.